app.js

```diff
@@ -235,30 +235,40 @@
                 message: message
             });
         } else {
-            User.register({
-                username: req.body.username
-            }, req.body.password, function(err, user) {
-                if (err) {
-                    console.log(err.message);
-                    var message = err.message;
-                    res.render("register", {
-                        message: message
-                    });
-                } else {
-                    User.updateOne({
-                        _id: user._id
-                    }, {
-                        email: req.body.email,
-                        fname: req.body.name,
-                        isAdmin: false
-                    }, function(err) {
-                        if (err) console.log(err);
-                    });
-                    passport.authenticate("local")(req, res, function() {
-                        res.redirect("/UserHome");
-                    });
-                }
-            });
+            var trimmedUsername = req.body.username.trim();
```

Comment:
Keep the `trim()` intact.
Additionally put a check for cases when there is spaces in between, i.e. username is of >1 words

For example: `abc def` this username should not be allowed
We must throw error to fill the form again.